src/components/CreatePost.js

```diff
@@ -0,0 +1,237 @@
+import React, {useState} from 'react';
+import {
+  Keyboard,
+  Modal,
+  StyleSheet,
+  Text,
+  TouchableWithoutFeedback,
+  View,
+} from 'react-native';
+import {APIPost} from '../utils/ApiClient';
+import {showToast} from '../utils/LocalStorage';
+import Button from './Button';
+import InputText from './InputText';
+import ErrorMessage from './ErrorMessage';
+import validate from './validator';
+import {FilePickerActionSheet} from './ActionSheet';
+import DashX from '@dashx/react-native';
+
+export const CreatePostModal = ({visible, dismissModal}) => {
+  const [newPost, setNewPost] = useState('');
+  const [requestInProgress, setRequestInProgress] = useState(false);
+  const [errorMessage, setErrorMessage] = useState({
+    newPost: false,
+  });
+  const [storePostImage, setStorePostImage] = useState();
+  const [storePostVideo, setStorePostVideo] = useState();
+  const [mediaType, setMediaType] = useState('');
+  const [showActionSheet, setShowActionSheet] = useState(false);
+
+  const validation = () => {
+    let count = 0;
+
+    for (let key in errorMessage) {
+      let validationResponse = validate(key, eval(key));
+      setErrorMessage(prev => {
+        return {
+          ...prev,
+          [key]: validationResponse,
+        };
+      });
+      if (validationResponse !== true) {
+        break;
+      } else {
+        count += 1;
+      }
+    }
+    if (count === Object.keys(errorMessage).length) {
+      createPost();
+    }
+  };
+
+  const createPost = async () => {
+    setRequestInProgress(true);
+    await APIPost({
+      endUrl: 'posts',
+      dataObject: {
+        text: newPost,
+        image: storePostImage?.postImage,
+        video: null,
+      },
+    });
+    setRequestInProgress(false);
+
+    showToast('Successfully Posted');
+
+    dismissModal?.(true);
+
+    // Reset post text
+    setNewPost();
+  };
+
+  const onPickPostImage = async pickedMedia => {
+    setShowActionSheet(false);
+
+    try {
+      if (mediaType === 'image') {
+        const response = await DashX.uploadExternalAsset(
+          pickedMedia,
+          'e8b7b42f-1f23-431c-b739-9de0fba3dadf',

```

Comment:
@Charan017 Please store these in a constants file and reference it from there.

Reply:
Done